sql/core/src/main/scala/org/apache/spark/sql/execution/python/ExtractPythonUDFs.scala

```diff
@@ -215,3 +228,49 @@ object ExtractPythonUDFs extends Rule[SparkPlan] with PredicateHelper {
     }
   }
 }
+
+
+/**
+ * Extract all the group aggregate Pandas UDFs in logical aggregation, evaluate the UDFs first
+ * and then the expressions that depend on the result of the UDFs.
+ */
+object ExtractGroupAggPandasUDFFromAggregate extends Rule[LogicalPlan] {
```

Comment:
I guess this rule name is not representing what this is doing.

Reply:
I end up removing the rule. Now I put the unsafe project inside `AggregateInPandasExec` similar to `HashAggregateExec`